cryptoutil/cryptoutil_test.go

```diff
@@ -0,0 +1,38 @@
+package cryptoutil
+
+import (
+	"testing"
+
+	"github.com/teamwork/test/diff"
+)
+
+const (
+	keyString         = "abcd1234abcd1234"
+	testPlaintextData = "HELLO WORLD"
+)
+
+func TestEncryptAndDecrypt(t *testing.T) {
+	tests := []string{
+		"HELLO WORLD",
+		"I love jam",
+		"HELLO WORLD",
+		"pls work already",
+		"haaaalp",
+	}
+
+	for _, testData := range tests {
+		data := []byte(testData)
+		cipher, err := Encrypt(keyString, data)
+		if err != nil {
+			t.Error(err)
```

Comment:
Small thing, and you don't really need to change it, but you probably want to use subtests; it's very easy to add (just wrap the inner body in `t.Run(func(t *testing.T) { })` and are quite useful.

Subtests are useful because they give you info about *which* test failed. Right now you don't see that, you just see that `TestEncryptAndDecrypt` failed on line 27. With subtests you see `TestEncryptAndDecrypt/that-one-bloody-test`, and you can use `go test -run TestEncryptAndDecrypt/that-one-bloody-test` to run just that one subtest.

You can also use `t.Fatal()` to stop the subtest after an initialisation error, like on this line. With `t.Error()` the test will continue, and if the initialisation failed here we already know that everything else will fail, so subsequent errors kind of become (possibly confusing/distracting) spam. It may also panic because `string(nil)` is a panic.

In this particular case it won't matter too much, because once this works it will probably stay working, but in cases where someone else breaks (and needs to fix) the test later on it's very useful, and will make the life of future maintainers a lot easier :-)

See [the docs](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks). You can also look at the other tests in this repo; almost all of them use subtests.